# HG changeset patch
# User John W. Eaton <jwe@octave.org>
# Date 1482965161 18000
#      Wed Dec 28 17:46:01 2016 -0500
# Branch stable
# Node ID 3793da0dbe3705a03cf74eb8ed4067ad647355e9
# Parent  cc025b4bd2fbd7a00f710960ec541f4d9c3cb6e7
fix cut and paste error in sparse matrix solver

* dSparse.cc (SparseMatrix::bsolve): Use DGBCON for solving general
banded matrix case.

diff --git a/liboctave/array/dSparse.cc b/liboctave/array/dSparse.cc
--- a/liboctave/array/dSparse.cc
+++ b/liboctave/array/dSparse.cc
@@ -5110,19 +5110,19 @@ SparseMatrix::bsolve (MatrixType &mattyp
               if (calc_cond)
                 {
                   char job = '1';
                   Array<double> z (dim_vector (3 * nr, 1));
                   double *pz = z.fortran_vec ();
                   Array<octave_idx_type> iz (dim_vector (nr, 1));
                   octave_idx_type *piz = iz.fortran_vec ();
 
-                  F77_XFCN (dpbcon, DPBCON,
+                  F77_XFCN (dgbcon, DGBCON,
                             (F77_CONST_CHAR_ARG2 (&job, 1),
-                             nr, n_lower, tmp_data, ldm,
+                             nc, n_lower, n_upper, tmp_data, ldm, pipvt,
                              anorm, rcond, pz, piz, err
                              F77_CHAR_ARG_LEN (1)));
 
                   if (err != 0)
                     err = -2;
 
                   volatile double rcond_plus_one = rcond + 1.0;
 
